service/service.py

```diff
@@ -349,6 +349,29 @@ def deliver_item(order_id, item_id):
 
     return make_response(jsonify(order.serialize()), status.HTTP_200_OK)
 
+######################################################################
+# DELIVER AN ORDER
+######################################################################
+@app.route("/orders/<int:order_id>/deliver", methods=["PUT"])
+def deliver_orders(order_id):
+    """ deliver all the items of the Order that have not being delivered yet """
+    app.logger.info("Request to deliver order with id: %s", order_id)
+    order = Order.find(order_id)
+    if not order:
+        raise NotFound("Order with id '{}' was not found.".format(order_id))
+
+    shipped_delivered_canceled_orders = 0
```

Comment:
You may want to change the name of this variable since it holds the count of the orders which have been either placed or cancelled.

Reply:
Ah yes. thank you.